src/Nethermind/Nethermind.Blockchain/Synchronization/SyncServer.cs

```diff
@@ -128,81 +145,91 @@ public void AddNewBlock(Block block, Node nodeWhoSentTheBlock)
             }
 
             if ((block.TotalDifficulty ?? 0) < _blockTree.BestSuggestedHeader.TotalDifficulty) return;
+            block.Header.TotalDifficulty = null; // important line - read the long comment above
 
+            // note that we only want to do that after updating peer information
             lock (_recentlySuggested)
             {
                 if (_recentlySuggested.Get(block.Hash) != null) return;
```

Comment:
All those returns from method make it very hard to get what the flow is.

Reply:
Agreed. I was thinking for a moment of extracting them into separate methods / some kind of chain of checks but decided it is a more detailed refactoring that would require 1h of work more. Cannot do that now.